confgenerator/confgenerator.go

```diff
@@ -191,12 +196,26 @@ func (uc *UnifiedConfig) generateOtelPipelines(ctx context.Context) (map[string]
 				}
 			}
 			for _, pID := range processorIDs {
-				// TODO: Change when we support trace processors.
-				processor, ok := m.Processors[pID]
+				var processor OTelProcessor
+				var ok bool
+				if pipelineType == "metrics" {
+					processor, ok = m.Processors[pID]
+				} else if pipelineType == "logs" {
+					var p LoggingProcessor
+					p, ok = l.Processors[pID]
+					if ok {
+						processor, ok = p.(OTelProcessor)
+					}
+				}
+				// TODO: Add trace processors?
 				if !ok {
 					return fmt.Errorf("processor %q not found", pID)
```

Comment:
`!ok` here can mean either "processor not found" or "processor is not an OTel processor"; can we distinguish the error message better?

Reply:
Yep, I added a better error message.